relay-server/src/services/processor.rs

```diff
@@ -663,6 +663,12 @@ struct ProcessEnvelopeState<'a, Group> {
 
     /// Reservoir evaluator that we use for dynamic sampling.
     reservoir: ReservoirEvaluator<'a>,
+
+    /// Track whether the event has already been fully normalized.
+    ///
+    /// If the processing pipeline applies changes to the event, it should
+    /// disable this flag to ensure the event is always normalized.
+    fully_normalized: bool,
```

Comment:
See `event_metrics_extracted` above.

```suggestion
    event_fully_normalized: bool,
```